aws/rust-runtime/aws-runtime/src/request_info.rs

```diff
@@ -63,16 +63,31 @@ impl RequestInfoInterceptor {
     }
 
     fn build_ttl_pair(&self, cfg: &ConfigBag) -> Option<(Cow<'static, str>, Cow<'static, str>)> {
-        let timeout_config = cfg.get::<TimeoutConfig>()?;
+        let timeout_config = cfg.timeout_config()?;
         let socket_read = timeout_config.read_timeout()?;
         let estimated_skew: Duration = cfg.get::<ServiceClockSkew>().cloned()?.into();
-        let current_time = SystemTime::now();
+        let current_time = cfg.time_source().now();
         let ttl = current_time.checked_add(socket_read + estimated_skew)?;
         let timestamp = DateTime::from(ttl);
-        let formatted_timestamp = timestamp
+        let mut formatted_timestamp = timestamp
             .fmt(Format::DateTime)
             .expect("the resulting DateTime will always be valid");
 
+        if let Some(start_of_fractional_secs) = formatted_timestamp.rfind('.') {
+            formatted_timestamp.truncate(start_of_fractional_secs)
```

Comment:
You can just set the subsecond nanos to 0 on the `DateTime` to avoid having to make assumptions about the date format.
```suggestion
            formatted_timestamp.truncate(start_of_fractional_secs)
```